framework/src/executioners/Transient.C

```diff
@@ -531,12 +540,15 @@ Transient::solveStep(Real input_dt)
 
   _problem.execute(EXEC_TIMESTEP_BEGIN);
 
-  if (_picard_max_its > 1)
+  if (_picard_max_its > 1 && _problem.hasMultiApps(EXEC_TIMESTEP_BEGIN))
```

Comment:
Potentially any aux kernel or user object of master on timestep_begin can change the residual evaluation of master. Unless we can find a rigorous check, we should drop this `&&` here unfortunately. We can provide an advanced input parameter `skip_timestep_begin_picard_norm_evaluation`.

Reply:
Another thinking is that when we add kernels, dgkernels or any objects for residual evaluation, if it couples aux variables or user objects, we can check mark a flag to indicate the kernel evaluation is affected by execution of timestep_begin, timestep_end, etc. Then based on this flag, we can bypass the norm evaluation here. Not sure if it is worthwhile to do this.

Reply:
Sorry, this is not going to work if we have something like this: kernel => material => auxkernel.

Reply:
I added a flag that will allow you to restore the existing behavior. Will that work for you? If I get a chance, I'll see if that fixes the tutorial failures in Rattlesnake.